src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -508,28 +676,65 @@ void CodegenNeuronCppVisitor::print_mechanism_register() {
     printer->add_line("_nrn_mechanism_register_data_fields(mech_type,");
     printer->increase_indent();
     const auto codegen_float_variables_size = codegen_float_variables.size();
+    const auto codegen_int_variables_size = codegen_int_variables.size();
     for (int i = 0; i < codegen_float_variables_size; ++i) {
         const auto& float_var = codegen_float_variables[i];
-        const auto print_comma = i < codegen_float_variables_size - 1 || info.emit_cvode;
+        const auto print_comma = i < codegen_float_variables_size - 1 ||
+                                 codegen_int_variables_size > 0 || info.emit_cvode;
         if (float_var->is_array()) {
-            printer->fmt_line("_nrn_mechanism_field<double>{{\"{}\", {}}} /* {} */{}",
-                              float_var->get_name(),
-                              float_var->get_length(),
-                              i,
-                              print_comma ? "," : "");
+            printer->fmt_line(
+                "_nrn_mechanism_field<double>{{\"{0}\", {1}}}{2} /* float var index {3} */",
+                float_var->get_name(),
+                float_var->get_length(),
+                print_comma ? "," : "",
+                i);
```

Comment:
`nitpick`: in such cases where its not obvious what the code or logic is, adding an example in code comment would be helpful? e.g.

````
we print the code of below form and we need to do x, y, z..........

 #define gnabar _ml->template fpfield<0>(_iml)
````

just as an additional context.

Reply:
Done